### PR TITLE
chore: add optional callback to Confirmation Modal cancel button

### PR DESCRIPTION
**Pull request by rssilva**
## Description of the change
Write out a good description of the change. A screenshot or video will also be helpful.
- Add onCancel optional callback to Confirmation Modal 

## Type of change

- [x] New feature (non-breaking change that adds functionality)

### Development

- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### CODE REVIEW

File: packages/components/src/components/ConfirmationModal/ConfirmationModal.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ export interface ConfirmationModalProps {
   confirmLabel?: string;
   destructive?: boolean;
   onConfirm: () => void;
+  onCancel?: () => void;
```
Comment (apvale) [inline packages/components/src/components/ConfirmationModal/ConfirmationModal.tsx:12]: Why do you think we need a callback for this?

Comment (rssilva) [reply packages/components/src/components/ConfirmationModal/ConfirmationModal.tsx:12]: To trigger actions (like track events, for example) when the user clicks on the cancel button

Comment (apvale) [reply packages/components/src/components/ConfirmationModal/ConfirmationModal.tsx:12]: I meant why changing this in the PR? Is there any specific scenario that would be necessary to override the state.hide action?

Comment (rssilva) [reply packages/components/src/components/ConfirmationModal/ConfirmationModal.tsx:12]: The state.hide() will still be called here

**Review by apvale — APPROVED** [review]

**Pull request merged**